Add expected input and output to integration docs

cbethell opened the pull request:
**Issue Addressed**
Closes #362 and #363

**What is the purpose of these changes?**
<!--Provide some background on the changes proposed.-->
This PR updates the integration module docs with expected input and output sections, as well as the additional parameters docs with the integration parameters.

**What changes did you make?**
<!--Describe the concrete changes that you made or additional features that were added, be as detailed as possible in the steps you took.-->
- Expected input and expected output sections were added to `optional-integration-analysis/README.md`
- An `Integration analysis parameters` section has been added to `additional-docs/additional-parameters.md`
- The `utils/sync-results.sh` script was updated to include the integration results and ran successfully to store the example results on AWS

**Any comments, concerns, or questions important for reviewers**
- Does there appear to be any important information missing from these updates?

**Checklist**

Place an `x` in all boxes that you have completed.

- [ ] I have run the most recent version of the code
- [ ] If I am adding in reports or generating any plots, I have reviewed the necessary reports
- [x] I have added all necessary documentation (if applicable)

allyhawkins commented on additional-docs/additional-parameters.md:
```diff
@@ -89,4 +90,21 @@ The following gene mapping parameters found in the `config/goi_config.yaml` file
+## Integration analysis parameters
+
+The [configuration file](https://snakemake.readthedocs.io/en/stable/snakefiles/configuration.html), `config/integration_config.yaml` sets the defaults for all parameters needed to run the data integration workflow.
+It is **not required** to alter these parameters to run the workflow, but if you would like to change the integration method(s) or the number of multi-processing threads to use, you can do so by changing these parameters via a text editor of your choice or at the command line per our documentation [here](./command-line-options.md). 
```
```suggestion
It is **not required** to alter these parameters to run the workflow.
If you would like to change the integration method(s) or the number of multi-processing threads to use, you can do so by changing these parameters via a text editor of your choice or at the command line per our documentation [here](./command-line-options.md). 
```
Just breaking this sentence up since it's a little long.

allyhawkins commented on additional-docs/additional-parameters.md:
```diff
@@ -89,4 +90,21 @@ The following gene mapping parameters found in the `config/goi_config.yaml` file
+| Parameter        | Description | Default value |
+|------------------|-------------|---------------|
+| `threads` | the number of multiprocessing threads to use | 1 |
+| `integration_method` | the method(s) to be used for integration | `"fastMNN,harmony"` |
```
```suggestion
| `integration_method` | the method(s) to be used for integration; to include multiple integration methods use a comma separated list. Currently, the workflow only supports `fastMNN` or `harmony`. | `"fastMNN,harmony"` |
```

allyhawkins commented on additional-docs/additional-parameters.md:
```diff
@@ -89,4 +90,21 @@ The following gene mapping parameters found in the `config/goi_config.yaml` file
+|------------------|-------------|---------------|
+| `threads` | the number of multiprocessing threads to use | 1 |
+| `integration_method` | the method(s) to be used for integration | `"fastMNN,harmony"` |
+| `batch_column` | the name of the `SingleCellExperiment` column that indicates the grouping of interest; this will generally be batches or cell types | `"library_id"` |
```
```suggestion
| `batch_column` | the name of the column in the `SingleCellExperiment` object indicating the original library each cell was derived from | `"library_id"` |
```
I don't think we want to mention cell types since we don't have full supporting right now for using cell types rather than batches for labeling and calculating metrics.

allyhawkins commented on additional-docs/additional-parameters.md:
```diff
@@ -89,4 +90,21 @@ The following gene mapping parameters found in the `config/goi_config.yaml` file
+| `threads` | the number of multiprocessing threads to use | 1 |
+| `integration_method` | the method(s) to be used for integration | `"fastMNN,harmony"` |
+| `batch_column` | the name of the `SingleCellExperiment` column that indicates the grouping of interest; this will generally be batches or cell types | `"library_id"` |
+| `cell_id_column` | the name of the `SingleCellExperiment` column that contains the cell barcodes | `"cell_id"` |
```
```suggestion
| `cell_id_column` | the name of the column in the `SingleCellExperiment` object containing the cell barcode | `"cell_id"` |
```

allyhawkins commented on optional-integration-analysis/README.md:
```diff
@@ -38,6 +41,16 @@ R 4.2 is required for running our pipeline, along with Bioconductor 3.15.
+
+To run this workflow, you will need to provide:
+
+1. Two or more RDS files containing normalized [output `SingleCellExperiment` objects](../README.md#expected-output) from the core dowstream analyses workflow or the processed `SingleCellExperiment` objects downloaded from the ScPCA portal (found in the `_processed.rds` file).
```
```suggestion
1. Two or more RDS files containing normalized [`SingleCellExperiment` objects](../README.md#expected-output).
These can be objects output from the core dowstream analyses workflow or the processed `SingleCellExperiment` objects downloaded from the ScPCA portal (found in the `_processed.rds` file).
```

allyhawkins commented on optional-integration-analysis/README.md:
```diff
@@ -98,3 +111,32 @@ If `--cores` is given without a number, all available cores are used to run the
+
+## Expected output
+
+For each provided `integration_group`, the workflow will return three files in the `results_dir` specified in the config file:
```
```suggestion
For each provided `integration_group`, the workflow will return three files saved in the `results_dir` specified in the config file:
```

allyhawkins commented on optional-integration-analysis/README.md:
```diff
@@ -98,3 +111,32 @@ If `--cores` is given without a number, all available cores are used to run the
+
+For each provided `integration_group`, the workflow will return three files in the `results_dir` specified in the config file:
+
+1. The `_merged_sce.rds` file containing the `SingleCellExperiment` object that results from merging the individual `SingleCellExperiment` objects specified in the project metadata.
```
```suggestion
1. The `<integration_group>_merged_sce.rds` file containing the `SingleCellExperiment` object that results from merging the individual `SingleCellExperiment` objects specified in the project metadata.
```
We are naming these with the integration group right? We should indicate that.

allyhawkins commented on optional-integration-analysis/README.md:
```diff
@@ -98,3 +111,32 @@ If `--cores` is given without a number, all available cores are used to run the
+For each provided `integration_group`, the workflow will return three files in the `results_dir` specified in the config file:
+
+1. The `_merged_sce.rds` file containing the `SingleCellExperiment` object that results from merging the individual `SingleCellExperiment` objects specified in the project metadata.
+2. The `_integrated_sce.rds` file containing the `SingleCellExperiment` object that has integrated data stored for each `integration_method` as specified in the project metadata.
```
Do we provide both the output of the merged and the integrated SCE? The integrated SCE contains the raw data but also the corrected embeddings right? If that's the case, we should actually remove saving the merged SCE file as an output and only output the integrated SCE.

allyhawkins commented on optional-integration-analysis/README.md:
```diff
@@ -98,3 +111,32 @@ If `--cores` is given without a number, all available cores are used to run the
+
+You can also download a ZIP file with an example of the output from running the data integration workflow, including the summary HTML report and the integrated `SingleCellExperiment` object stored as an RDS file, [here](https://scpca-references.s3.amazonaws.com/example-data/scpca-downstream-analyses/integration_example_results.zip).
+
+### What to expect in the output `SingleCellExperiment` object
```
```suggestion
### What to expect in the integrated `SingleCellExperiment` object
```

allyhawkins commented on optional-integration-analysis/README.md:
```diff
@@ -98,3 +111,32 @@ If `--cores` is given without a number, all available cores are used to run the
+
+### What to expect in the output `SingleCellExperiment` object
+
+In the [`reducedDim`](https://bioconductor.org/books/3.13/OSCA.intro/the-singlecellexperiment-class.html#dimensionality-reduction-results) of the integrated output `SingleCellExperiment` object, you can find the following:
```
```suggestion
In the [`reducedDim`](https://bioconductor.org/books/3.13/OSCA.intro/the-singlecellexperiment-class.html#dimensionality-reduction-results) of the integrated `SingleCellExperiment` object, you can find the following:
```

allyhawkins commented on optional-integration-analysis/README.md:
```diff
@@ -98,3 +111,32 @@ If `--cores` is given without a number, all available cores are used to run the
+
+In the [`reducedDim`](https://bioconductor.org/books/3.13/OSCA.intro/the-singlecellexperiment-class.html#dimensionality-reduction-results) of the integrated output `SingleCellExperiment` object, you can find the following:
+
+- Integrated results stored in the reduced dimensions and named using the associated type of integration performed.
```
Do we also provide UMAP? Or just PCA?

cbethell replied:
We do also provide UMAP! Will add a note here

allyhawkins reviewed (commented):
This looks mostly good. Can you also include a new section in the [main Readme for additional modules](https://github.com/AlexsLemonade/scpca-downstream-analyses/tree/cbethell/add-expected-input-integration-docs#additional-analysis-modules) for the integration module? That might make sense to include as a separate PR.

cbethell commented:
Thanks for the feedback @allyhawkins! I removed that `merged_sce` file from the outputs and the mention of it from the docs. Also incorporated the remaining comments you left above -- will include a new section in the main README in an upcoming PR!

allyhawkins commented on optional-integration-analysis/README.md:
```diff
@@ -10,9 +10,12 @@ The merged objects contain the raw and normalized counts for all included librar
 - [Configure config file](#configure-config-file)
 - [Running the workflow](#running-the-workflow)
+- [Expected output](#expected-output)
+  - [What to expect in the output `SingleCellExperiment` object](#what-to-expect-in-the-output-singlecellexperiment-object)
```
Make sure to re-run doctoc.

allyhawkins commented on optional-integration-analysis/README.md:
```diff
@@ -98,3 +112,31 @@ If `--cores` is given without a number, all available cores are used to run the
+
+For each provided `integration_group`, the workflow will return two files saved in the `results_dir` specified in the config file:
+
+1. The `<integration_group>_integrated_sce.rds` file containing the `SingleCellExperiment` object that has integrated data stored for each `integration_method` as specified in the project metadata.
```
```suggestion
1. The `<integration_group>_integrated_sce.rds` file containing the `SingleCellExperiment` object containing batch-corrected embeddings for each `integration_method` specified in the project metadata.
```

allyhawkins commented on optional-integration-analysis/README.md:
```diff
@@ -98,3 +112,31 @@ If `--cores` is given without a number, all available cores are used to run the
+
+In the [`reducedDim`](https://bioconductor.org/books/3.13/OSCA.intro/the-singlecellexperiment-class.html#dimensionality-reduction-results) of the integrated `SingleCellExperiment` object, you can find the following:
+
+- Integrated results stored in the reduced dimensions and named using the associated type of integration performed.
```
``` suggestion
Batch-corrected embeddings stored in the reduced dimensions and named using the associated type of integration performed.
```
The data from integration that we are using is batch-corrected PCA's so we should be explicit when referring to what we mean by "data integration results". Also since the only thing we provide are the PCA/UMAP, I don't think we need a bullet point here.

allyhawkins reviewed (approved):
I have a few more suggestions on cleaning up the text. Once those are incorporated and doctoc is run again then this is good to go and I don't need to see it again.
